### PR TITLE
Added WPS Download handler - refactored

### PR DESCRIPTION
**Pull request by pmbohm**

### CODE REVIEW

**Comment by pmbohm** [comment]
This can replace https://github.com/aodn/aodn-portal/pull/1989
I had trouble finishing tests for WpsDownloadHandler.js

**Comment by jkburges** [comment]
This seems to be exactly the same as #1989 - closing this since there are already comments over on the other PR.

I won't delete the branch, just in case, but @pmbohm please do if it's not needed.

**Comment by pmbohm** [comment]
@jkburges Heres the actual changes I selfishly kept on my computer for the last 4 days.
@danfruehauf

**Comment by jkburges** [comment]
This still looks the same to me, although I haven't actually done a `git diff` this time.  I'm was working off #1989 (integrating the client and server) so let me know if you want me to go off this one.  Close this one or the other, up to you.

**Comment by pmbohm** [comment]
@jkburges Tests fixed now
@danfruehauf

**Comment by pmbohm** [comment]
@jkburges It has new files! 
Portal.cart.AsyncDownloadHandler etc

**Comment by jkburges** [comment]
> It has new files!

Yeah, it does now, but not when I made the comment!

**Comment by jkburges** [comment]
@pmbohm note adef63e69f39219ae16980  I have temporarily disabled the WPS download option.  I will take care of re-enabling when I finish the client/server integration (coming soon, but can be on a separate PR).

File: grails-app/conf/Config.groovy [hunk 1]
```diff
@@ -268,7 +268,8 @@ portal {
             "WWW:LINK-1.0-http--link",
             "WWW:DOWNLOAD-1.0-http--download",
             "WWW:DOWNLOAD-1.0-http--downloaddata",
-            "WWW:LINK-1.0-http--downloaddata"
+            "WWW:LINK-1.0-http--downloaddata",
+            "OGC:WPS"
```
Comment (jkburges) [inline grails-app/conf/Config.groovy]: Isn't this a `dataFile` rather than a `webPage`?

File: src/test/javascript/spec_helper.js [hunk 2]
```diff
@@ -191,3 +191,8 @@ var textToXML = function(text) {
     catch (e) {
     }
 };
+
+var jsonFromUrl = function(url, urlStart) {
+    var jobParameters = url.substring(urlStart.length);
+    return Ext.util.JSON.decode(decodeURIComponent(jobParameters));
+};
```
Comment (jkburges) [inline src/test/javascript/spec_helper.js]: EOL

**Comment by jkburges** [comment]
Just the one comment, then GTG :+1:

**Comment by pmbohm** [comment]
@jkburges New pull request up

**Pull request merged**